cpp/src/arrow/json/parser-benchmark.cc

```diff
@@ -133,49 +126,41 @@ static void BenchmarkJSONReading(benchmark::State& state,  // NOLINT non-const r
 
     std::shared_ptr<Table> table;
     ABORT_NOT_OK(reader->Read(&table));
-
-    if (table->num_rows() != num_rows) {
-      std::cerr << "Parsing incomplete\n";
-      std::abort();
-    }
-
-    tables[read_options.use_threads] = table;
   }
-  state.SetBytesProcessed(state.iterations() * json.size());
 
-  if (tables[false] && tables[true]) {
-    AssertTablesEqual(*tables[false], *tables[true]);
-  }
+  state.SetBytesProcessed(state.iterations() * json.size());
 }
 
-static void BM_ReadJSONBlockWithSchema(
-    benchmark::State& state) {  // NOLINT non-const reference
-  const int32_t num_rows = 50000;
+static void BenchmarkReadJSONBlockWithSchema(
+    benchmark::State& state, bool use_threads) {  // NOLINT non-const reference
+  const int32_t num_rows = 500000;
   auto read_options = ReadOptions::Defaults();
-  read_options.use_threads = state.range(0);
+  read_options.use_threads = use_threads;
 
   auto parse_options = ParseOptions::Defaults();
   parse_options.unexpected_field_behavior = UnexpectedFieldBehavior::Error;
-  parse_options.explicit_schema = schema({field("int", int32()), field("str", utf8())});
+  parse_options.explicit_schema = TestSchema();
 
-  std::default_random_engine engine;
-  std::string json;
-  for (int i = 0; i < num_rows; ++i) {
-    StringBuffer sb;
-    Writer writer(sb);
-    ABORT_NOT_OK(Generate(parse_options.explicit_schema, engine, &writer));
-    json += sb.GetString();
-    json += "\n";
-  }
+  auto json = TestJsonData(num_rows);
   BenchmarkJSONReading(state, json, num_rows, read_options, parse_options);
 }
 
-BENCHMARK(BM_ReadJSONBlockWithSchema)
-    ->MinTime(1.0)
-    ->Unit(benchmark::kMicrosecond)
-    ->Arg(true)
-    ->Arg(false)
-    ->UseRealTime();
+static void ReadJSONBlockWithSchemaSingleThread(
+    benchmark::State& state) {  // NOLINT non-const reference
+  BenchmarkReadJSONBlockWithSchema(state, false);
+}
+
+static void ReferenceReadJSONBlockWithSchemaMultiThread(
```

Comment:
I'm curious, why is this a reference benchmark?

Reply:
I decided not to include it yet because the n_proc inference isn't accounting for cpuset (cgroup). Thus I fear this benchmark is highly variable if it runs in cgroups, which I've prepared on dgx2 for benchmarks.

Since we don't have a database, I'll revert this and leave it as a normal benchmark.